File: backend/src/main/java/com/codeloom/backend/dao/SubmissionRepository.java
package com.codeloom.backend.dao;

import com.codeloom.backend.model.Submission;
import java.util.*;
import org.springframework.data.repository.CrudRepository;

public interface SubmissionRepository extends CrudRepository<Submission, UUID> {
  Collection<Submission> findByUserIdAndProblemId(UUID userId, long problemId);
}

File: backend/src/main/kotlin/com/codeloom/backend/dao/SubmissionRepository.kt
package com.codeloom.backend.dao

import com.codeloom.backend.model.Submission
import org.springframework.data.repository.CrudRepository
import java.util.Collection
import java.util.UUID

interface SubmissionRepository : CrudRepository<Submission, UUID> {
    fun findByUserIdAndProblemId(
        userId: UUID,
        problemId: Long,
    ): Collection<Submission>
}
